### ISSUE TITLE
Inventory dropping broken by 4.8 update 

### ISSUE THREAD

**Issue by BadMagic100** [fifty-six/HollowKnight.QoL#45]
On latest updates, opening inventory never prevents the hard fall animation/sound effect, and only prevents functional hard falls if the inventory is open when you hit the ground.

**Comment by fifty-six** [comment]
wdym? The actual opening of the inventory isn't what cancels the hard fall, it's the *closing* which gives control back and thus "cancels" the hard fall by allowing you to move anyways.

**Comment by BadMagic100** [comment]
It used to be the case that having the inventory open cancels your hardfall timer, and you would therefore be able to close your inventory early and still not get a hardfall. On looking further it seems like this is actually an intentional change in b17767ddeed3287ce4babc46189e4cad0bd1c757, but I'm not clear why. The old behavior is preferred by myself and several rando players who pointed this out to me. Some folks have said that maybe the current behavior is consistent with what happens on 1221 which I guess would make it correct but annoying, but I haven't gotten solid confirmation on this

**Comment by fifty-six** [comment]
Yeah, the idea was to be closer to how it acts on the old patches. I could add an option to revert that back if people want, though I think the default should probably be closer to the old patches given the general idea of the "SpeedBroke" stuff is to restore behavior from older patches. You just gotta close inventory as you hit the ground to regain control instead

**Comment by BadMagic100** [comment]
Yeah, I think having an option to toggle the behavior would be nice, but I do agree that in general the default behavior should be to replicate the behavior on older patches of HK

**Comment by fifty-six** [comment]
Alright, I have finals and such next week so I don't know exactly when I'll get around to it, but I'll make this an option.
